mesh/nurbs.cpp

```diff
@@ -129,19 +129,25 @@ void KnotVector::Print(std::ostream &os) const
    knot.Print(os, knot.Size());
 }
 
-
 void KnotVector::PrintFunctions(std::ostream &os, int samples) const
 {
    Vector shape(Order+1);
 
    double x, dx = 1.0/double (samples - 1);
 
-   for (int i = 0; i <GetNE() ; i++)
+   /* @a i is a counter including elements between repeated knots if
+      present. This is required for usage of CalcShape. */
+   int i = 0;
```

Comment:
As a matter of style, I would prefer a slightly longer name than `i` for a variable with scope outside the loop, e.g. `cnt`.